pymarlin/core/trainer.py

```diff
@@ -140,11 +143,38 @@ def __init__(
         self.trainer_backend.init(self._get_trainer_backend_args())
         self.trainer_backend.update_state(self.checkpointed_states.trainer_backend_state)
 
+        if self.args.ort: 
+            from torch_ort import ORTModule 
+
+            assert(hasattr(self, 'model') and isinstance(self.model, torch.nn.module), "expected self.model property of type torch.nn.module")
+            
+            self.logger.info("Converting to ORTModule ....") 
+            self.module = ORTModule(self.module)
+
+            self.module.on_begin_train_epoch = self.module._module_metadata.original_module.on_begin_train_epoch
+            self.module.on_end_train_epoch = self.module._module_metadata.original_module.on_end_train_epoch
+            self.module.on_end_train = self.module._module_metadata.original_module.on_end_train
+            self.module.get_val_dataloaders = self.module._module_metadata.original_module.get_val_dataloaders
+            self.module.get_train_dataloader = self.module._module_metadata.original_module.get_train_dataloader
+            self.module.get_state = self.module._module_metadata.original_module.get_state
+            self.module.get_optimizers_schedulers = self.module._module_metadata.original_module.get_optimizers_schedulers
+            self.module.on_end_backward = self.module._module_metadata.original_module.on_end_backward
+            self.module.on_end_train_step = self.module._module_metadata.original_module.on_end_train_step
+            self.module.on_end_val_epoch = self.module._module_metadata.original_module.on_end_val_epoch
+
+
     def train(self):
-        """ Train and validate the model"""
+        """ Train and validate the model"""    
+        start_train_stable_time = 0
```

Comment:
Remove boilerplate code like logging runtime. Calculating time itself is adding 7 lines of extra code. Better approach is to create decorators.